src/charm.py

```diff
@@ -112,12 +114,12 @@ def _on_storage_detaching(self, event: StorageDetachingEvent) -> None:
 
         # in the case where there may be replication recovery may be possible
         if self.peer_relation and len(self.peer_relation.units):
-            message = "manual partition reassignment suggested due to potential log data loss"
-            logger.warning(f"Removing storage - {message}")
+            message = "manual partition reassignment from replicated brokers recommended due to lost partitions on removed storage volumes"
+            logger.warning(f"removing storage - {message}")
             self.unit.status = BlockedStatus(message)
         else:
-            message = "potential log-data loss"
-            logger.error(f"Removing storage - {message}")
+            message = "potential log-data loss due to storage removal without replication"
```

Comment:
I think you're right and we can address it when it's 'docs' time. I was wondering whether it would be possible to actually restrict allowing a storage removal in that case, but I don't think we have access to that at the charm level.